go/vt/vttablet/tabletserver/tabletserver_test.go

```diff
@@ -2154,17 +2154,138 @@ func TestReserveStats(t *testing.T) {
 	assert.NotEmpty(t, tsv.te.txPool.env.Stats().UserReservedTimesNs.Counts()["test"])
 }
 
-func setupTabletServerTest(t *testing.T) (*fakesqldb.DB, *TabletServer) {
+func TestDatabaseNameReplaceByKeyspaceName(t *testing.T) {
+	db, tsv := setupTabletServerTest(t, "keyspaceName")
+	db.SetName("databaseInMysql")
+	defer tsv.StopService()
+	defer db.Close()
+
+	executeSQL := "select * from test_table limit 1000"
+	executeSQLResult := &sqltypes.Result{
+		Fields: []*querypb.Field{
+			{
+				Type:     sqltypes.VarBinary,
+				Database: "databaseInMysql",
+			},
+		},
+		RowsAffected: 1,
+		Rows: [][]sqltypes.Value{
+			{sqltypes.NewVarBinary("row01")},
+		},
+	}
+	r := db.AddQuery(executeSQL, executeSQLResult)
+	require.NotNil(t, r)
+	target := tsv.sm.target
+
+	// Testing Execute Method
+	transactionID, _, err := tsv.Begin(ctx, &target, nil)
+	require.NoError(t, err)
+	res, err2 := tsv.Execute(ctx, &target, executeSQL, nil, transactionID, 0, &querypb.ExecuteOptions{IncludedFields: querypb.ExecuteOptions_ALL})
+	require.NoError(t, err2)
+	for _, field := range res.Fields {
+		require.Equal(t, "keyspaceName", field.Database)
+	}
+	_, err = tsv.Commit(ctx, &target, transactionID)
+	require.NoError(t, err)
+
+	// Testing StreamExecute Method
+	callback := func(res *sqltypes.Result) error {
+		for _, field := range res.Fields {
+			if field.Database != "" {
+				require.Equal(t, "keyspaceName", field.Database)
+			}
+		}
+		return nil
+	}
+	if err := tsv.StreamExecute(ctx, &target, executeSQL, nil, 0, &querypb.ExecuteOptions{IncludedFields: querypb.ExecuteOptions_ALL}, callback); err != nil {
+		t.Fatalf("TabletServer.StreamExecute should success: %s, but get error: %v",
```

Comment:
instead of using `if err` and `t.Fatalf`, it's more readable if if you use the `require.NoError(t,err)` pattern